src/realm/object-store/property.hpp

```diff
@@ -336,6 +344,15 @@ inline Property::Property(std::string name, PropertyType type, std::string objec
 {
 }
 
+inline Property::Property(std::string name, const std::vector<PropertyType>& nested_types, std::string public_name)
+    : name(std::move(name))
+    , public_name(std::move(public_name))
+    , nested_types(nested_types)
+{
+    REALM_ASSERT(!nested_types.empty());
+    type = nested_types.front();
```

Comment:
I guess the type should be that at the back. It should be the type of the leaf array where the actual values are stored.

Reply:
ok, I did this initially, but then I thought that was strange to have a property that is nested whose actual type is (Int or whatever is the leaf type). I am going to revet the code in order to follow your suggestion.